js/src/logger.ts

```diff
@@ -1911,33 +1920,7 @@ export function traced<IsAsyncFlush extends boolean = false, R = void>(
   callback: (span: Span) => R,
   args?: StartSpanArgs & SetCurrentArg & AsyncFlushArg<IsAsyncFlush>
 ): PromiseUnless<IsAsyncFlush, R> {
-  const { span, isLogger } = ((): { span: Span; isLogger: boolean } => {
```

Comment:
Unrelated cleanup: I noticed that the lower-level `startSpan` function was not looking at the `parent` arg, so I factored out the bit of logic that considers it and call it in both places.